### ISSUE TITLE
Performance issue with skip

### ISSUE THREAD

**Issue by dlichtenberger** [kolmodin/binary#73]
I'm using the lazy interface of Data.Binary.Get to parse large binaries from disk, where most parts are skipped on the first pass. When using skip directly, I get a severe performance and space usage problem when the skipped byte count is large (many megabytes). I'm not an expert on lazy bytestring internals, but it seems like the input data is held on for too long before being skipped ("PINNED" memory usage in -hc heap profile). Using this wrapper around skip makes it several orders of magnitude faster and does not explode on the heap (I'm using GHC 7.10.1):

```haskell
import qualified Data.Binary.Get as G

skipMany :: Int -> G.Get ()
skipMany bytes = 
  replicateM_ rep (G.skip cs) >> G.skip rest
 where
   cs = 1024
   (rep, rest) = bytes `quotRem` cs
```

**Comment by kolmodin** [comment]
Right. The current implementation of `skip` concatenates the bytes to be skipped, and then throws it away.

This is unnecessarily expensive. I'll work on a more efficient implementation.

**Comment by kolmodin** [comment]
I might add that even with the more efficient implementation, it will still force to read all the data from disk (or wherever it comes from) which might be inefficient for your case.
The most efficient implementation would be to `seek` in the file descriptor and only read the parts needed.

**Comment by kolmodin** [comment]
Fixed in e46b7b61111579befcc99954be8a0f48738f8ef7.
